client/state/redux-store.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * External Dependencies
+ */
+import { Reducer, Store } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { addReducerToStore, WithAddReducer } from './add-reducer';
+
+type QueueEntry = [ string[], Reducer ];
+
+let applicationStore: ( Store & WithAddReducer ) | undefined;
+let reducerRegistrationQueue: QueueEntry[] = [];
+
+export function setStore( store: Store & WithAddReducer ) {
+	// Only set store once.
+	if ( applicationStore ) {
+		return;
+	}
+
+	applicationStore = store;
+	// Synchronously add all pending reducers.
+	for ( const [ key, reducer ] of reducerRegistrationQueue ) {
+		addReducerToStore( applicationStore )( key, reducer );
+	}
+	// Clear registration queue to avoid dangling references.
+	reducerRegistrationQueue = [];
+}
+
+export function getStore() {
```

Comment:
I added this method because I tend to think symmetrically, but I'm not seeing a good use for it anymore. Should I remove it?

Reply:
did you answer your own question?

Reply:
I suppose I was looking for validation more than an answer 🙂 I'll take your comment as one of the former 😄